test/e2e/networkpolicy_test.go

```diff
@@ -806,6 +806,163 @@ func TestIngressPolicyWithoutPortNumber(t *testing.T) {
+	createAgnhostPodOnNodeWithMultiPort := func(name string, nodeName string) error {
+		var containers []corev1.Container
+		for _, port := range serverPorts {
+			containers = append(containers, makeContainerSpec(port))
+		}
+		podSpec := corev1.PodSpec{
+			Containers:    containers,
+			RestartPolicy: corev1.RestartPolicyNever,
+			HostNetwork:   false,
+		}
+		if nodeName != "" {
+			podSpec.NodeSelector = map[string]string{
+				"kubernetes.io/hostname": nodeName,
+			}
+		}
+		if nodeName == controlPlaneNodeName() {
+			// tolerate NoSchedule taint if we want Pod to run on control-plane Node
+			noScheduleToleration := controlPlaneNoScheduleToleration()
+			podSpec.Tolerations = []corev1.Toleration{noScheduleToleration}
+		}
+		pod := &corev1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: name,
+				Labels: map[string]string{
+					"antrea-e2e": name,
+					"app":        getImageName(agnhostImage),
+				},
+			},
+			Spec: podSpec,
+		}
+		if _, err := data.clientset.CoreV1().Pods(testNamespace).Create(context.TODO(), pod, metav1.CreateOptions{}); err != nil {
+			return err
+		}
+		return nil
+	}
+
+	serverName, serverIPs, cleanupFunc := createAndWaitForPod(t, data, createAgnhostPodOnNodeWithMultiPort, "test-server-", "")
+	defer cleanupFunc()
+
+	clientName, _, cleanupFunc := createAndWaitForPod(t, data, data.createBusyboxPodOnNode, "test-client-", "")
+	defer cleanupFunc()
+
+	preCheck := func(serverIP string) {
+		// The client can connect to server on all ports.
+		for _, port := range serverPorts {
+			if err = data.runNetcatCommandFromTestPod(clientName, serverIP, port); err != nil {
+				t.Fatalf("Pod %s should be able to connect %s, but was not able to connect", clientName, net.JoinHostPort(serverIP, fmt.Sprint(port)))
+			}
+		}
+
+	}
+
+	if clusterInfo.podV4NetworkCIDR != "" {
+		preCheck(serverIPs.ipv4.String())
+	}
+	if clusterInfo.podV6NetworkCIDR != "" {
+		preCheck(serverIPs.ipv6.String())
+	}
+
+	protocol := corev1.ProtocolTCP
+	spec := &networkingv1.NetworkPolicySpec{
+		PodSelector: metav1.LabelSelector{
+			MatchLabels: map[string]string{
+				"antrea-e2e": serverName,
+			},
+		},
+		PolicyTypes: []networkingv1.PolicyType{networkingv1.PolicyTypeIngress},
+		Ingress: []networkingv1.NetworkPolicyIngressRule{
+			{
+				Ports: []networkingv1.NetworkPolicyPort{
+					{
+						Protocol: &protocol,
+						Port:     &intstr.IntOrString{Type: intstr.Int, IntVal: policyPort},
+						EndPort:  &policyEndPort,
+					},
+				},
+				From: []networkingv1.NetworkPolicyPeer{{
+					PodSelector: &metav1.LabelSelector{
+						MatchLabels: map[string]string{
+							"antrea-e2e": clientName,
+						},
+					}},
+				},
+			},
+		},
+	}
+	np, err := data.createNetworkPolicy("test-networkpolicy-ingress-with-endport", spec)
+	if err != nil {
+		t.Fatalf("Error when creating network policy: %v", err)
+	}
+	defer func() {
+		if err = data.deleteNetworkpolicy(np); err != nil {
+			t.Fatalf("Error when deleting network policy: %v", err)
```

Comment:
as a rule you should avoid using `Fatalf` in cleanup functions

Reply:
Done.